### ISSUE TITLE
`possibly-used-before-assignment` does not treat `assert_never` the same when imported from `typing_extensions`

### ISSUE THREAD

**Issue by Kakadus** [pylint-dev/pylint#9780]
### Bug description

```python
The following code succeeds in the following environment using python 3.12:


pylint 3.2.5
astroid 3.2.2
Python 3.12.4 (main, Jun  7 2024, 06:33:07) [GCC 14.1.1 20240522]
```

but fails with python 3.10

```
pylint 3.2.5
astroid 3.2.2
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
```

```py
from enum import Enum
from typing import Literal
from typing_extensions import assert_never


class As(Enum):
    A = "A"
    AA = "AA"


a: Literal[As.A, As.AA] = As.A

if a == As.A:
    NUM_AS = 1
elif a == As.AA:
    NUM_AS = 2
else:
    assert_never(a)

print(NUM_AS)
```

### Command used

```shell
./venv/bin/pylint --disable=all --enable=possibly-used-before-assignment test.py
```

### Pylint output

```python
either nothing (success) or

test.py:20:6: E0606: Possibly using variable 'NUM_AS' before assignment (possibly-used-before-assignment)
```

### Expected behavior

I expect pylint to detect that everything is right and it should not detect any issues.

Also, I expect pylint to behave consistent accross different python version

### Pylint version

```shell
pylint 3.2.5
```

### OS / Environment

Arch (with python 3.12)
Ubuntu (with python 3.10)

**Comment by jacobtylerwalls** [comment]
Thanks for the report, we forgot to handle typing_extensions
